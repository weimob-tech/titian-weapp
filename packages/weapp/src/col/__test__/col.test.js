const simulate = require('miniprogram-simulate');
const { getComponent, getComponentId } = require('../../common/test/getComponent');

describe('test component', () => {
  const col = getComponentId('col');
  const row = getComponentId('row');

  it('should render success', () => {
    const comp = getComponent('col');
    expect(comp).toMatchSnapshot();
  });

  it('basic', () => {
    const comp = simulate.render(
      simulate.load({
        usingComponents: {
          'ti-row': row,
          'ti-col': col
        },
        template: `
        <view>
          <ti-row>
            <ti-col span='{{8}}'>
              <view>1</view>
            </ti-col>
            <ti-col span='{{8}}'>
              <view>2</view>
            </ti-col>
            <ti-col span='{{8}}'>
              <view>3</view>
            </ti-col>
          </ti-row>
        </view>
      `
      })
    );
    comp.attach(document.createElement('parent-wrapper'));
    expect(comp.toJSON()).toMatchSnapshot();
  });
});
