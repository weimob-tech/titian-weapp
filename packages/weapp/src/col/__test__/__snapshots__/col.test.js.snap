// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component basic 1`] = `
<main>
  <wx-view>
    <main>
      <wx-view
        class="titian-row-flex titian-row"
        style="margin-left: -0rpx; margin-right: -0rpx;"
      >
        <main>
          <wx-view
            class="ext-class titian-col-8 titian-col-flex-8 titian-col-offset-0 titian-col"
            style="padding-left:0rpx;padding-right:0rpx"
          >
            <wx-view>
              1
            </wx-view>
          </wx-view>
        </main>
        <main>
          <wx-view
            class="ext-class titian-col-8 titian-col-flex-8 titian-col-offset-0 titian-col"
            style="padding-left:0rpx;padding-right:0rpx"
          >
            <wx-view>
              2
            </wx-view>
          </wx-view>
        </main>
        <main>
          <wx-view
            class="ext-class titian-col-8 titian-col-flex-8 titian-col-offset-0 titian-col"
            style="padding-left:0rpx;padding-right:0rpx"
          >
            <wx-view>
              3
            </wx-view>
          </wx-view>
        </main>
      </wx-view>
    </main>
  </wx-view>
</main>
`;

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="ext-class titian-col-0 titian-col-flex-0 titian-col-offset-0 titian-col"
    style="padding-left:0rpx;padding-right:0rpx"
  />
</main>
`;
